bundles/org.openhab.binding.linky/src/main/java/org/openhab/binding/linky/internal/handler/LinkyHandler.java

```diff
@@ -0,0 +1,387 @@
+import okhttp3.FormBody.Builder;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+
+/**
+ * The {@link LinkyHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Gaël L'hopital - Initial contribution
+ */
+
+@NonNullByDefault
+public class LinkyHandler extends BaseThingHandler {
+    private final Logger logger = LoggerFactory.getLogger(LinkyHandler.class);
+
+    private static final String LOGIN_BASE_URI = "https://espace-client-connexion.enedis.fr/auth/UI/Login";
+    private static final String API_BASE_URI = "https://espace-client-particuliers.enedis.fr/group/espace-particuliers/suivi-de-consommation";
+    private static final DateTimeFormatter API_DATE_FORMAT = DateTimeFormatter.ofPattern("dd/MM/yyyy");
+    private static final Builder LOGIN_BODY_BUILDER = new FormBody.Builder().add("encoded", "true")
+            .add("gx_charset", "UTF-8").add("SunQueryParamsString",
+                    Base64.getEncoder().encodeToString("realm=particuliers".getBytes(StandardCharsets.UTF_8)));
+
+    private static final int REFRESH_FIRST_HOUR_OF_DAY = 5;
+    private static final int REFRESH_INTERVAL_IN_MIN = 360;
+
+    private final OkHttpClient client = new OkHttpClient.Builder().followRedirects(false)
+            .cookieJar(new LinkyCookieJar()).build();
+    private final Gson gson = new Gson();
+
+    private @NonNullByDefault({}) ScheduledFuture<?> refreshJob;
+    private final WeekFields weekFields;
+
+    private ExpiringDayCache<LinkyConsumptionData> cachedDaylyData;
+    private ExpiringDayCache<LinkyConsumptionData> cachedMonthlyData;
+    private ExpiringDayCache<LinkyConsumptionData> cachedYearlyData;
+
+    public LinkyHandler(Thing thing, LocaleProvider localeProvider) {
+        super(thing);
+        this.weekFields = WeekFields.of(localeProvider.getLocale());
+        this.cachedDaylyData = new ExpiringDayCache<LinkyConsumptionData>("daily cache", REFRESH_FIRST_HOUR_OF_DAY,
+                () -> {
+                    final LocalDate today = LocalDate.now();
+                    return getConsumptionData(DAILY, today.minusDays(13), today, true);
+                });
+        this.cachedMonthlyData = new ExpiringDayCache<LinkyConsumptionData>("monthly cache", REFRESH_FIRST_HOUR_OF_DAY,
+                () -> {
+                    final LocalDate today = LocalDate.now();
+                    return getConsumptionData(MONTHLY, today.withDayOfMonth(1).minusMonths(1), today, true);
+                });
+        this.cachedYearlyData = new ExpiringDayCache<LinkyConsumptionData>("yearly cache", REFRESH_FIRST_HOUR_OF_DAY,
+                () -> {
+                    final LocalDate today = LocalDate.now();
+                    return getConsumptionData(YEARLY, LocalDate.of(today.getYear() - 1, 1, 1), today, true);
+                });
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing Linky handler.");
```

Comment:
please call `updateStatus(ThingStatus.UNKNOWN);` just in case something goes horribly wrong with the login that the Thing will be in a valid state.